go/cmd/dolt/commands/batsee.go

```diff
@@ -0,0 +1,377 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/fatih/color"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/cmd/dolt/errhand"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env"
+	"github.com/dolthub/dolt/go/libraries/utils/argparser"
+)
+
+var batseeDoc = cli.CommandDocumentationContent{
+	ShortDesc: `Run the Bats Tests concurrently`,
+	LongDesc:  `From within the integration-test/bats directory, run the bats tests concurrently. Output for each test is written to a file in the batsee_output directory.`,
+	Synopsis: []string{
+		`-t 42`,
+		`--skip-slow --max-time 1h15m`,
+		`--retries 2 --only types.bats,foreign-keys.bats`,
+	},
+}
+
+type BatseeCmd struct {
+	messages []string
+}
+
+var _ cli.Command = &BatseeCmd{}
+var _ cli.RepoNotRequiredCommand = &BatseeCmd{}
+
+func (b BatseeCmd) Name() string {
+	return "batsee"
+}
+
+func (b BatseeCmd) Description() string {
+	return batseeDoc.ShortDesc
+}
+func (b BatseeCmd) Docs() *cli.CommandDocumentation {
+	return cli.NewCommandDocumentation(batseeDoc, b.ArgParser())
+}
+
+func (b BatseeCmd) Hidden() bool {
+	return true
+}
+
+func (b BatseeCmd) ArgParser() *argparser.ArgParser {
+	ap := argparser.NewArgParserWithMaxArgs(b.Name(), 0)
+	ap.SupportsUint("threads", "t", "threads", "Number of tests to execute in parallel. Defaults to 12")
```

Comment:
The flag names should be defined as constants at the top of the file, like

```
const (
    ThreadsFlag = "threads"
)

ap.SupportsUint(ThreadsFlag, ...)

threads, hasThreads : = apr.GetValue(ThreadsFlag)
```

Reply:
I'm soooooo tepid to this suggestion :-)